packages/tsutils/src/hashStringToInt.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+/**
+ * Convert a string to an integer hash
+ * Kudos: https://rclayton.silvrback.com/distributed-locking-with-postgres-advisory-locks
+ */
+export const hashStringToInt = (str: string) => {
+  let hash = 5381;
+  let i = -1;
+  while (i < str.length - 1) {
+    i += 1;
+    // eslint-disable-next-line no-bitwise
+    hash = (hash * 33) ^ str.charCodeAt(i);
+  }
+  // eslint-disable-next-line no-bitwise
+  return hash >>> 0;
+};
```

Comment:
Ah `@tupaia/tsutils`... Originally I had imagined this package would be for typescript specific code (eg. type guards, useful generic types, etc.)

However, at this point it really seems like it's just the same as `@tupaia/utils`, and honestly I'd much rather create new utils code in typescript than in javascript.

I've created a [ticket](https://linear.app/bes/issue/RN-639/convert-tupaiautils-to-typescript-and-merge-with-tupaiatsutils) to merge these two packages into a single `@tupaia/utils` package which is written in typescript

Reply:
This kind of small but dense algorithm is the sort of thing I like unit tests around. Not sure if we even care about the expected output, but perhaps we care that e.g. it's a somewhat sensible output, the same input always hashes to the same int, and we don't get conflicts between similar inputs? Totally not required if we trust the source of the algo :-)

Reply:
Hehehe I had actually been toying with how to test this in the back of my mind for a while.

The properties we want the function to hold are:
a) It will convert any string into a 64bit integer
b) It will always convert the same string into the same integer

It's the kind of function that would really benefit from a concept I've explored before, [property based testing](https://medium.com/criteo-engineering/introduction-to-property-based-testing-f5236229d237). 

It might give that a try, otherwise a simple 'generate random string, hash it a bunch of a times and ensure they're all equal' should do the trick? 😉